gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -133,9 +131,18 @@ func (o *options) validate() error {
 	// validate and parse bucket list
 	o.bucketAliases = bucketAliases{}
 	for _, bucket := range o.allowedBuckets {
-		if err := o.parseBucket(bucket); err != nil {
+		prowPath, err := o.parseBucket(bucket)
+		if err != nil {
 			return err
 		}
+
+		if o.provider == "" {
+			o.provider = prowPath.StorageProvider()
+		} else if o.provider != prowPath.StorageProvider() {
+			// If GCS buckets are served, we create a GCS-only client in getStorageClient, hence we cannot serve S3 buckets in
+			// this case.
+			return fmt.Errorf("serving buckets of different storage providers at the same time is not supported")
+		}
```

Comment:
This new constraint will be backwards incompatible, right?

Reply:
Yes, and this constraint should have been added in https://github.com/kubernetes/test-infra/pull/31364 right away. 
Before https://github.com/kubernetes/test-infra/pull/31364, one could only use GCS, so considering https://github.com/kubernetes/test-infra/pull/31364 and this PR as one single change means its compatible with earlier versions of gcsweb.

Reply:
Apparently wrong link here as well.
Your explanation sounds good to me assuming the breaking change was recent enough that no one would have adopted it (since they would now be broken by this change).

Reply:
I also meant https://github.com/kubernetes/test-infra/pull/31241 in this comment.

Exactly, the breaking change was pretty recent and folks running the typical gcsweb setup (public bucket + anonymous auth) on GCP were broken by it.